fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBDatabaseFactory.java

```diff
@@ -533,6 +558,37 @@ public Executor newContextExecutor(@Nullable Map<String, String> mdcContext) {
         return newExecutor;
     }
 
+    /**
+     * Set the number of threads per FDB client version. The default value is 1.
+     *
+     * @param threadsPerClientV the number of threads per client version. Cannot be less than 1.
+     *
+     * @deprecated Call directly on {@link FDBDatabaseFactoryImpl}. This will be removed in a future release.
+     */
+    @Deprecated
+    public static void setThreadsPerClientVersion(int threadsPerClientV) {
+        if (staticOptionsSet) {
```

Comment:
Minor, but wouldn't this have been simpler as just:

```
     @Deprecated
     public static void setThreadsPerClientVersion(int threadsPerClientV) {
         FDBDatabaseFactoryImpl.instance().setThreadsPerClientVersion(threadsPerClientV);
     }
```

Reply:
That would be a loop, `FDBDatabaseFactoryImpl.instance().setThreadsPerClientVersion()` calls this with:

```
    public static void setThreadsPerClientVersion(int threadsPerClientV) {
        FDBDatabaseFactory.setThreadsPerClientVersion(threadsPerClientV);
    }
```

Reply:
Sorry, my implication was to move the static variable and its surrounding logic to `FDBDatabaseFactoryImpl`, then have these deprecated calls simply call out to them since, when these deprecated methods are removed, they will just need to be moved into `Impl` anyway.

Reply:
Ah gotcha